layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-mender/tegra-state-scripts/files/redundant-boot-install-script

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
+
+mnt=
+LABELCHARS="AB"
+COPY_MACHINE_ID=@COPY_MACHINE_ID@
+
+cleanup() {
+    [ -n "$mnt" ] || return
+    for d in sys proc dev run; do
+	if mountpoint -q "${mnt}/${d}"; then
+	    umount "${mnt}/${d}" >/dev/null 2>&1 || true
+	fi
+    done
+    if mountpoint -q "$mnt"; then
+        umount "$mnt" >/dev/null 2>&1 || true
+    fi
+    rmdir "$mnt" >/dev/null 2>&1 || true
+}
+
+echo "Installing NVIDIA bootloader update payload"
+
+num_slots=`nvbootctrl get-number-slots`
+if [ $num_slots != 2 ]; then
+    echo "Enabling A/B update mode using nv_update_engine"
+    nv_update_engine --enable-ab
+    if [ ! "$?" -eq 0 ]; then
+        echo "ERR: could not enable A/B update by using nv_update_engine" >&2
+        exit 1
+    fi
+fi
+current_slot=`nvbootctrl get-current-slot`
+echo "Current boot slot: $current_slot"
+otherslot=`expr 1 - $current_slot`
+cfglbl="\"RootfsPart${LABELCHARS:$otherslot:1}\""
+devnam=`grep -h "$cfglbl:" /etc/mender/mender.conf /var/lib/mender/mender.conf | cut -d: -f2 | cut -d, -f1 | tr -d '" '`
+if [ -z "$devnam" ]; then
+    echo "ERR: could not determine device name for boot slot $otherslot" >&2
+    exit 1
+fi
+mnt=`mktemp -d -t nvbup.XXXXXX`
+if [ -z "$mnt" -o ! -d "$mnt" ]; then
+    echo "ERR: could not create directory for mounting install partition" >&2
+    exit 1
+fi
+mount "$devnam" "$mnt"
```

Comment:
@madisongh would you prefer we don't set `-o ro` here or [for u-boot](https://github.com/mendersoftware/meta-mender-community/pull/208/files#diff-7a8bb63a7208e096d44d2a7523e16f71ff9f85e2b6a12d62e4696abc21c8d034R38) ?  I think it's not strictly required to mount as read-only as long as we don't end up writing the rootfs as a part of ota update.  It might help catch problems with ota update tools which assume the root partition is writable, however it could also create problems for platforms which don't need to ensure the root partition is unmodified during OTA (which I think would only be an issue for those who need delta update support).

Reply:
Right, mounting readonly might be a good idea here, since the stock `nv_update_engine` is apt to write a file into the `/opt/ota_package` directory, which would be undesirable for the delta-update case.  I don't think it will fail if it can't write that file.

If someone is going to replace the bootloader update tools used by this script with something else and really needs the rootfs to be writable, they can override this script with another one.